### PR TITLE
Fix loading of stencils using `{% extends %}`

### PR DESCRIPTION
**Pull request by devyhan**
Resolves [#3406](https://github.com/tuist/tuist/issues/3406)

### Short description 📝

- Expect the Stencils directory to be present in the file inspector when it is defined and filled with stencil.
- Parent template be able to `{% extensions %}` tag.
- The version of `StencilSwiftKit` has been modified to 2.9.0 to inject loader into `stencilSwiftEnvironment`.

### How to test the changes locally 🧐

After downloading this [sample project](https://github.com/tuist/tuist/files/7115065/StencilBug.zip), determine if this [#3406](https://github.com/tuist/tuist/issues/3406) PR has been resolved.

### Checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] The title of the PR will be used as changelog entry, please make sure it is clear and suitable
- [ ] In case the PR introduces changes that affect users, the documentation has been updated
- [ ] Contributors have checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`

### CODE REVIEW

File: Sources/TuistGraph/Models/PluginStencil.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import Foundation
+import TSCBasic
+
+/// Stencil plugin model
+public struct PluginStencil: Equatable {
```
Comment (danyf90) [inline Sources/TuistGraph/Models/PluginStencil.swift:5]: Why does this need to be in `TuistGraph` module?

File: Tuist/Dependencies.swift [hunk 2]
```diff
@@ -16,7 +16,7 @@ let dependencies = Dependencies(
             .package(url: "https://github.com/Flight-School/AnyCodable", from: "0.6.6"),
             .package(url: "https://github.com/rnine/Checksum.git", .upToNextMajor(from: "1.0.2")),
             .package(url: "https://github.com/stencilproject/Stencil.git", .upToNextMajor(from: "0.15.1")),
-            .package(url: "https://github.com/SwiftGen/StencilSwiftKit.git", .exact("2.8.0")),
+            .package(url: "https://github.com/SwiftGen/StencilSwiftKit.git", .exact("2.9.0")),
```
Comment (danyf90) [inline Tuist/Dependencies.swift:19]: Please update it in `Package.swift` as well 🙏

File: Sources/TuistScaffold/TemplateGenerator.swift [hunk 3]
```diff
@@ -115,10 +116,11 @@ public final class TemplateGenerator: TemplateGenerating {
                     context: attributes
                 )
             case let .file(path):
+                let injectedLoaderEnvironment = stencilSwiftEnvironment(templatePaths: [Path(path.dirname)])
                 let fileContents = try FileHandler.shared.readTextFile(path)
                 // Render only files with .stencil extension
                 if path.extension == "stencil" {
-                    renderedContents = try environment.renderTemplate(
+                    renderedContents = try injectedLoaderEnvironment.renderTemplate(
```
Comment (danyf90) [inline Sources/TuistScaffold/TemplateGenerator.swift:123]: Is this change related to `fail to load stencils using {% extends %}`? Can we split the 2 fixes in 2 PRs so we can have 2 changelog entries?

**Review by danyf90 — CHANGES_REQUESTED** [review]

File: Sources/TuistGraph/Models/PluginStencil.swift (hunk 1 above)
Comment (devyhan) [reply Sources/TuistGraph/Models/PluginStencil.swift:5]: I looked again today and it seems to be a module that has nothing to do with `TuistGraph` and has something to do with `TuistLoader`. 

Can i move that module to `TuistLoader/Models`?

File: Tuist/Dependencies.swift (hunk 2 above)
Comment (devyhan) [reply Tuist/Dependencies.swift:19]: Thanks 🙏

File: Sources/TuistScaffold/TemplateGenerator.swift (hunk 3 above)
Comment (danyf90) [reply Sources/TuistScaffold/TemplateGenerator.swift:123]: Thanks 🚀

**Comment by devyhan** [comment]
I need help @danyf90 🥲

image

CI build fails. 
The simulator with that ID doesn't seem to exist. 
Could it be related to macOS-12 in this recently changed [workflow file](https://github.com/tuist/tuist/commit/5abe84dd0e95431f4cebcdaaf32c941d97328084#:~:text=%3A%20macos%2D11-,runs%2Don%3A%20macos%2D12,-needs%3A%20prepare)?

**Supported list on macOS-12**
image

**Supported list on macOS-11**
image

**Comment by danyf90** [comment]
> I need help @danyf90 🥲
> 
> image
> 
> CI build fails. The simulator with that ID doesn't seem to exist. Could it be related to macOS-12 in this recently changed [workflow file](https://github.com/tuist/tuist/commit/5abe84dd0e95431f4cebcdaaf32c941d97328084#:~:text=%3A%20macos%2D11-,runs%2Don%3A%20macos%2D12,-needs%3A%20prepare)?
> 
> **Supported list on macOS-12** image
> 
> **Supported list on macOS-11** image
> 
> How can i fix it?

I know, all PRs are failing in the past couple of days, but I didn't had the time to look at it 🤔 

Might be that, don't care about it in this PR, we'll try to fix it in a separate PR (if you would like to have a look at it, feel free to open a separate PR 🙏

**Pull request merged**